How to activate the plugin für groovy or javascript only projects

the plugin is only worky if you have at least on java file in your setup.

We have JavaScript and also groovy only projects - in this case you can't activate the plugin.

Thanks for the plugin, great work.

@Janpopan unfortunately the workaround becomes a bit of a rabbit hole.  For my NodeJS project:

- I create a temporary dummy Java file as part of the CI build process, right before running Sonar.
- I ensure that I have a Java Sonar quality profile which includes the two `licensecheck` rules, and nothing else, and assign my project to this quality profile.  See screenshot (note I have redacted the project names to anonymise this)
![Screenshot 2021-03-02 at 10 16 15](https://user-images.githubusercontent.com/6017680/109633845-7b331200-7b40-11eb-9ab6-b1e7ba305a70.png)
- Since I have real Java projects, I use FindBugs etc on them; you can't disable the FindBugs exporter on a per-profile basis, even if you have no FindBugs rules
- FindBugs complains that my Java file is not compiled and crashes, meaning there is no Sonar output:

```
ERROR: Error during SonarQube Scanner execution
java.lang.IllegalStateException: Can not execute Findbugs
	at org.sonar.plugins.findbugs.FindbugsExecutor.execute(FindbugsExecutor.java:188)
	at org.sonar.plugins.findbugs.FindbugsSensor.execute(FindbugsSensor.java:114)
	at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:48)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:85)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.lambda$execute$1(ModuleSensorsExecutor.java:59)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.withModuleStrategy(ModuleSensorsExecutor.java:77)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:59)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:82)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:137)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:123)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:393)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:389)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:358)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:137)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:123)
	at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:144)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:137)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:123)
	at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:72)
	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:66)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:185)
	at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:137)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:112)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:75)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
Caused by: java.lang.IllegalStateException: One (sub)project contains Java source files that are not compiled (/var/lib/jenkins/workspace/REDACTED).
	at org.sonar.plugins.findbugs.FindbugsConfiguration.getFindbugsProject(FindbugsConfiguration.java:123)
	at org.sonar.plugins.findbugs.FindbugsExecutor.execute(FindbugsExecutor.java:119)
	... 31 more
```

I'm looking into inserting _compiled_ versions of the dummy Java code in my NodeJS project as well, but this is fast becoming a complete rabbit hole.

Can't we instead change the definition of the licensecheck rules defined in the plugin, so they can be applied to non-Java projects?  That seems like the correct thing to do, rather than this painful workaround.

--- a/src/main/java/at/porscheinformatik/sonarqube/licensecheck/Dependency.java
+++ b/src/main/java/at/porscheinformatik/sonarqube/licensecheck/Dependency.java
@@ -1,5 +1,7 @@
 package at.porscheinformatik.sonarqube.licensecheck;
 
+import static at.porscheinformatik.sonarqube.licensecheck.LicenseCheckRulesDefinition.LANG_JAVA;
+
 import java.io.StringReader;
 import java.io.StringWriter;
 import java.util.ArrayList;
@@ -19,15 +21,22 @@ public class Dependency implements Comparable<Dependency>
     private String name;
     private String version;
     private String license;
+    private String lang;
     private Status status;
     private String pomPath;
 
-    public Dependency(String name, String version, String license)
+    public Dependency(String name, String version, String license, String lang)
     {
         super();
         this.name = name;
         this.version = version;
         this.license = license;
+        this.lang = lang;
+    }
+
+    public Dependency(String name, String version, String license)
+    {
+        this(name, version, license, LANG_JAVA);
     }
 
     public String getName()
@@ -58,6 +67,16 @@ public class Dependency implements Comparable<Dependency>
     public void setLicense(String license)
     {
         this.license = license;
+    }
+
+    public String getLang()
+    {
+        return lang;
+    }
+
+    public void setLang(String lang)
+    {
+        this.lang = lang;
     }
 
     public void setStatus(final Status status)
@@ -140,7 +159,7 @@ public class Dependency implements Comparable<Dependency>
                         JsonObject dependencyJson = dependenciesJson.getJsonObject(i);
                         dependencies.add(
                             new Dependency(dependencyJson.getString("name"), dependencyJson.getString("version"),
-                                dependencyJson.getString("license")));
+                                dependencyJson.getString("license"), dependencyJson.getString("lang", LANG_JAVA)));
                     }
                 }
             }
@@ -155,7 +174,7 @@ public class Dependency implements Comparable<Dependency>
                     String name = subParts.length > 0 ? subParts[0] : null;
                     String version = subParts.length > 1 ? subParts[1] : null;
                     String license = subParts.length > 2 ? subParts[2] : null;
-                    dependencies.add(new Dependency(name, version, license));
+                    dependencies.add(new Dependency(name, version, license, LANG_JAVA));
                 }
             }
         }
@@ -177,6 +196,7 @@ public class Dependency implements Comparable<Dependency>
             generator.write("name", dependency.getName());
             generator.write("version", dependency.getVersion());
             generator.write("license", license != null ? license : " ");
+            generator.write("lang", dependency.getLang());
             generator.writeEnd();
         }
         generator.writeEnd();

--- a/src/main/java/at/porscheinformatik/sonarqube/licensecheck/LicenseCheckMetrics.java
+++ b/src/main/java/at/porscheinformatik/sonarqube/licensecheck/LicenseCheckMetrics.java
@@ -9,11 +9,6 @@ import org.sonar.api.measures.Metrics;
 
 public class LicenseCheckMetrics implements Metrics
 {
-    public static final String LICENSE_CHECK_KEY = "licensecheck";
-
-    public static final String LICENSE_CHECK_UNLISTED_KEY = "licensecheck.unlisted";
-    public static final String LICENSE_CHECK_NOT_ALLOWED_LICENSE_KEY = "licensecheck.notallowedlicense";
-
     public static final String LICENSE_CHECK_DEPENDENCY_KEY = "licensecheck.dependency";
     private static final String LICENSE_CHECK_LICENSE_KEY = "licensecheck.license";
 

--- a/src/main/java/at/porscheinformatik/sonarqube/licensecheck/LicenseCheckRulesDefinition.java
+++ b/src/main/java/at/porscheinformatik/sonarqube/licensecheck/LicenseCheckRulesDefinition.java
@@ -1,19 +1,3 @@
-/*
- * SonarQube Licencecheck Plugin
- * Copyright (C) 2016 Porsche Informatik
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
 package at.porscheinformatik.sonarqube.licensecheck;
 
 import org.sonar.api.rule.Severity;
@@ -24,24 +8,43 @@ import org.sonar.api.server.rule.RulesDefinition;
  */
 public final class LicenseCheckRulesDefinition implements RulesDefinition
 {
+    public static final String LANG_JAVA = "java";
+    public static final String LANG_JS = "js";
+    public static final String LANG_GROOVY = "grvy";
+
+    public static final String RULE_REPO_KEY = "licensecheck";
+    public static final String RULE_REPO_KEY_JS = "licensecheck-js";
+    public static final String RULE_REPO_KEY_GROOVY = "licensecheck-groovy";
+
+    public static final String RULE_UNLISTED_KEY = "licensecheck.unlisted";
+    public static final String RULE_NOT_ALLOWED_LICENSE_KEY = "licensecheck.notallowedlicense";
+
     @Override
     public void define(Context context)
     {
-        NewRepository repository = context.createRepository(LicenseCheckMetrics.LICENSE_CHECK_KEY, "java");
-        repository.setName("License Check");
+        NewRepository[] repos = new NewRepository[]{
+            context.createRepository(RULE_REPO_KEY, LANG_JAVA),
+            context.createRepository(RULE_REPO_KEY_JS, LANG_JS),
+            context.createRepository(RULE_REPO_KEY_GROOVY, LANG_GROOVY)
+        };
 
-        repository
-            .createRule(LicenseCheckMetrics.LICENSE_CHECK_UNLISTED_KEY)
-            .setName("Dependency has unknown license [license-check]")
-            .setHtmlDescription("The dependencies license could not be determined!")
-            .setSeverity(Severity.BLOCKER);
+        for (NewRepository repo : repos)
+        {
+            repo.setName("License Check");
 
-        repository
-            .createRule(LicenseCheckMetrics.LICENSE_CHECK_NOT_ALLOWED_LICENSE_KEY)
-            .setName("License is not allowed [license-check]")
-            .setHtmlDescription("Violation because the license of the dependency is not allowed.")
-            .setSeverity(Severity.BLOCKER);
+            repo
+                .createRule(RULE_UNLISTED_KEY)
+                .setName("Dependency has unknown license [license-check]")
+                .setHtmlDescription("The dependencies license could not be determined!")
+                .setSeverity(Severity.BLOCKER);
 
-        repository.done();
+            repo
+                .createRule(RULE_NOT_ALLOWED_LICENSE_KEY)
+                .setName("License is not allowed [license-check]")
+                .setHtmlDescription("Violation because the license of the dependency is not allowed.")
+                .setSeverity(Severity.BLOCKER);
+
+            repo.done();
+        }
     }
 }

--- a/src/main/java/at/porscheinformatik/sonarqube/licensecheck/LicenseCheckSensor.java
+++ b/src/main/java/at/porscheinformatik/sonarqube/licensecheck/LicenseCheckSensor.java
@@ -1,5 +1,9 @@
 package at.porscheinformatik.sonarqube.licensecheck;
 
+import static at.porscheinformatik.sonarqube.licensecheck.LicenseCheckRulesDefinition.RULE_REPO_KEY;
+import static at.porscheinformatik.sonarqube.licensecheck.LicenseCheckRulesDefinition.RULE_REPO_KEY_GROOVY;
+import static at.porscheinformatik.sonarqube.licensecheck.LicenseCheckRulesDefinition.RULE_REPO_KEY_JS;
+
 import java.util.Set;
 import java.util.TreeSet;
 import java.util.concurrent.ConcurrentHashMap;
@@ -111,7 +115,7 @@ public class LicenseCheckSensor implements Sensor
     public void describe(SensorDescriptor descriptor)
     {
         descriptor.name("License Check")
-            .createIssuesForRuleRepository(LicenseCheckMetrics.LICENSE_CHECK_KEY);
+            .createIssuesForRuleRepositories(RULE_REPO_KEY, RULE_REPO_KEY_JS, RULE_REPO_KEY_GROOVY);
     }
 
     @Override

--- a/src/main/java/at/porscheinformatik/sonarqube/licensecheck/ValidateLicenses.java
+++ b/src/main/java/at/porscheinformatik/sonarqube/licensecheck/ValidateLicenses.java
@@ -156,8 +156,7 @@ public class ValidateLicenses
 
         NewIssue issue = context
             .newIssue()
-            .forRule(RuleKey.of(LicenseCheckMetrics.LICENSE_CHECK_KEY,
-                LicenseCheckMetrics.LICENSE_CHECK_NOT_ALLOWED_LICENSE_KEY));
+            .forRule(RuleKey.of(getRepoKey(dependency), LicenseCheckRulesDefinition.RULE_NOT_ALLOWED_LICENSE_KEY));
         issue.at(issue.newLocation().on(context.project()).message(
             "Dependency " + dependency.getName() + " uses a not allowed license " + dependency.getLicense()));
         issue.save();
@@ -171,11 +170,24 @@ public class ValidateLicenses
 
         NewIssue issue = context
             .newIssue()
-            .forRule(RuleKey.of(LicenseCheckMetrics.LICENSE_CHECK_KEY,
-                LicenseCheckMetrics.LICENSE_CHECK_UNLISTED_KEY));
+            .forRule(RuleKey.of(getRepoKey(dependency), LicenseCheckRulesDefinition.RULE_UNLISTED_KEY));
         issue.at(issue.newLocation().on(context.project())
             .message("No License found for Dependency: " + dependency.getName()));
         issue.save();
+    }
+
+    private static String getRepoKey(Dependency dependency)
+    {
+        switch (dependency.getLang())
+        {
+            case LicenseCheckRulesDefinition.LANG_JS:
+                return LicenseCheckRulesDefinition.RULE_REPO_KEY_JS;
+            case LicenseCheckRulesDefinition.LANG_GROOVY:
+                return LicenseCheckRulesDefinition.RULE_REPO_KEY_GROOVY;
+            case LicenseCheckRulesDefinition.LANG_JAVA:
+            default:
+                return LicenseCheckRulesDefinition.RULE_REPO_KEY;
+        }
     }
 
     private static boolean contains(String[] items, String valueToFind)

--- a/src/main/java/at/porscheinformatik/sonarqube/licensecheck/gradle/GradleDependencyScanner.java
+++ b/src/main/java/at/porscheinformatik/sonarqube/licensecheck/gradle/GradleDependencyScanner.java
@@ -1,25 +1,29 @@
 package at.porscheinformatik.sonarqube.licensecheck.gradle;
 
-import at.porscheinformatik.sonarqube.licensecheck.Dependency;
-import at.porscheinformatik.sonarqube.licensecheck.Scanner;
-import at.porscheinformatik.sonarqube.licensecheck.mavenlicense.MavenLicenseService;
-
-import org.codehaus.plexus.util.StringUtils;
-import org.sonar.api.utils.log.Logger;
-import org.sonar.api.utils.log.Loggers;
-
-import javax.json.Json;
-import javax.json.JsonArray;
-import javax.json.JsonObject;
-import javax.json.JsonReader;
-
-import java.io.*;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+import org.codehaus.plexus.util.StringUtils;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+import at.porscheinformatik.sonarqube.licensecheck.Dependency;
+import at.porscheinformatik.sonarqube.licensecheck.LicenseCheckRulesDefinition;
+import at.porscheinformatik.sonarqube.licensecheck.Scanner;
+import at.porscheinformatik.sonarqube.licensecheck.mavenlicense.MavenLicenseService;
 
 public class GradleDependencyScanner implements Scanner
 {
@@ -83,7 +87,7 @@ public class GradleDependencyScanner implements Scanner
                 moduleLicenseUrl = arrModuleUrls.getString(0, null);
             }
             Dependency dep = new Dependency(jsonDepObj.getString("moduleName", null),
-                jsonDepObj.getString("moduleVersion", null), moduleLicense);
+                jsonDepObj.getString("moduleVersion", null), moduleLicense, LicenseCheckRulesDefinition.LANG_JAVA);
             dep.setPomPath(moduleLicenseUrl);
             dependencySet.add(dep);
         }

--- a/src/main/java/at/porscheinformatik/sonarqube/licensecheck/maven/LicenseFinder.java
+++ b/src/main/java/at/porscheinformatik/sonarqube/licensecheck/maven/LicenseFinder.java
@@ -12,8 +12,8 @@ import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
 import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
 
-
 import at.porscheinformatik.sonarqube.licensecheck.Dependency;
+import at.porscheinformatik.sonarqube.licensecheck.LicenseCheckRulesDefinition;
 
 class LicenseFinder
 {
@@ -40,7 +40,8 @@ class LicenseFinder
                 {
                     Parent parent = model.getParent();
                     Dependency dependency =
-                        new Dependency(parent.getGroupId() + ":" + parent.getArtifactId(), parent.getVersion(), null);
+                        new Dependency(parent.getGroupId() + ":" + parent.getArtifactId(), parent.getVersion(), null,
+                            LicenseCheckRulesDefinition.LANG_JAVA);
                     return getLicenses(DirectoryFinder.getPomPath(dependency,
                         DirectoryFinder.getMavenRepsitoryDir(userSettings, globalSettings)),
                         userSettings, globalSettings);

--- a/src/main/java/at/porscheinformatik/sonarqube/licensecheck/maven/MavenDependencyScanner.java
+++ b/src/main/java/at/porscheinformatik/sonarqube/licensecheck/maven/MavenDependencyScanner.java
@@ -28,6 +28,7 @@ import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
 
 import at.porscheinformatik.sonarqube.licensecheck.Dependency;
+import at.porscheinformatik.sonarqube.licensecheck.LicenseCheckRulesDefinition;
 import at.porscheinformatik.sonarqube.licensecheck.Scanner;
 import at.porscheinformatik.sonarqube.licensecheck.mavendependency.MavenDependency;
 import at.porscheinformatik.sonarqube.licensecheck.mavendependency.MavenDependencyService;
@@ -185,7 +186,8 @@ public class MavenDependencyScanner implements Scanner
             path = path.substring(0, lastDotIndex) + ".pom";
         }
 
-        Dependency dependency = new Dependency(groupId + ":" + artifactId, version, null);
+        Dependency dependency =
+            new Dependency(groupId + ":" + artifactId, version, null, LicenseCheckRulesDefinition.LANG_JAVA);
         if (new File(path).exists())
         {
             dependency.setPomPath(path);

--- a/src/main/java/at/porscheinformatik/sonarqube/licensecheck/npm/PackageJsonDependencyScanner.java
+++ b/src/main/java/at/porscheinformatik/sonarqube/licensecheck/npm/PackageJsonDependencyScanner.java
@@ -19,6 +19,7 @@ import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
 
 import at.porscheinformatik.sonarqube.licensecheck.Dependency;
+import at.porscheinformatik.sonarqube.licensecheck.LicenseCheckRulesDefinition;
 import at.porscheinformatik.sonarqube.licensecheck.Scanner;
 
 public class PackageJsonDependencyScanner implements Scanner
@@ -113,7 +114,7 @@ public class PackageJsonDependencyScanner implements Scanner
                     else if (packageJson.containsKey("licenses"))
                     {
                         final JsonArray licenses = packageJson.getJsonArray("licenses");
-                        if (licenses.size() == 1) 
+                        if (licenses.size() == 1)
                         {
                             license = licenses.getJsonObject(0).getString("type", "");
                         }
@@ -135,7 +136,8 @@ public class PackageJsonDependencyScanner implements Scanner
                         }
                     }
 
-                    dependencies.add(new Dependency(packageName, packageJson.getString("version", null), license));
+                    dependencies.add(new Dependency(packageName, packageJson.getString("version", null), license,
+                        LicenseCheckRulesDefinition.LANG_JS));
 
                     if (resolveTransitiveDeps)
                     {

--- a/src/main/web/dashboard/dependencies.vue
+++ b/src/main/web/dashboard/dependencies.vue
@@ -6,7 +6,8 @@
       <thead>
         <tr>
           <th v-for="dependency in columns" v-bind:key="dependency" v-on:click="sort(dependency)" scope="col"> {{dependency}} 
-            <div class="arrow" v-if="dependency == sortByDep" v-bind:class="{ 'arrow_up' : sortDirectionDep === 'asc', 'arrow_down' : sortDirectionDep === 'desc'}"></div>
+            <div class="arrow" v-if="dependency === sortByDep"
+                 v-bind:class="{ 'arrow_up' : sortDirectionDep === 'asc', 'arrow_down' : sortDirectionDep === 'desc'}"></div>
           </th>
         </tr>
       </thead>
@@ -47,16 +48,13 @@ export default {
       return this.dependencies.sort((a, b) => {
         let modifier = 1;
         if (this.sortDirectionDep === "desc") modifier = -1;
-        if (a[this.sortByDep] < b[this.sortByDep]) return -1 * modifier;
-        if (a[this.sortByDep] > b[this.sortByDep]) return 1 * modifier;
+        if (a[this.sortByDep] < b[this.sortByDep]) return -modifier;
+        if (a[this.sortByDep] > b[this.sortByDep]) return modifier;
         return 0;
       });
     },
     "columns": function columns() {
-      if (this.dependencies.length == 0) {
-        return [];
-      }
-      return Object.keys(this.dependencies[0])
+      return ["name", "version", "license", "status"];
     }
   },
   methods: {

--- a/src/test/java/at/porscheinformatik/sonarqube/licensecheck/DependencyTest.java
+++ b/src/test/java/at/porscheinformatik/sonarqube/licensecheck/DependencyTest.java
@@ -12,8 +12,8 @@ import org.junit.Test;
 public class DependencyTest
 {
     private static final String DEPENDENCIES_JSON =
-        "[{\"name\":\"another\",\"version\":\"2.1.0\",\"license\":\"MIT\"}," +
-            "{\"name\":\"library\",\"version\":\"1.0.0\",\"license\":\"Apache-2.0\"}]";
+        "[{\"name\":\"another\",\"version\":\"2.1.0\",\"license\":\"MIT\",\"lang\":\"java\"}," +
+            "{\"name\":\"library\",\"version\":\"1.0.0\",\"license\":\"Apache-2.0\",\"lang\":\"java\"}]";
     private static final String DEPENDENCIES_STRING = "another~2.1.0~MIT;library~1.0.0~Apache-2.0";
     private static final Dependency DEP1 = new Dependency("another", "2.1.0", "MIT");
     private static final Dependency DEP2 = new Dependency("library", "1.0.0", "Apache-2.0");

--- a/src/test/java/at/porscheinformatik/sonarqube/licensecheck/LicenseCheckRulesDefinitionTest.java
+++ b/src/test/java/at/porscheinformatik/sonarqube/licensecheck/LicenseCheckRulesDefinitionTest.java
@@ -1,0 +1,24 @@
+package at.porscheinformatik.sonarqube.licensecheck;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.junit.Test;
+import org.sonar.api.server.rule.RulesDefinition;
+
+public class LicenseCheckRulesDefinitionTest
+{
+    @Test
+    public void define()
+    {
+        RulesDefinition.Context context = new RulesDefinition.Context();
+
+        new LicenseCheckRulesDefinition().define(context);
+
+        assertThat(context.repositories().size(), is(3));
+        for (RulesDefinition.Repository repository : context.repositories())
+        {
+            assertThat(repository.rules().size(), is(2));
+        }
+    }
+}

--- a/src/test/java/at/porscheinformatik/sonarqube/licensecheck/ValidateLicensesTest.java
+++ b/src/test/java/at/porscheinformatik/sonarqube/licensecheck/ValidateLicensesTest.java
@@ -8,7 +8,6 @@ import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
-import java.io.File;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashSet;
@@ -67,7 +66,7 @@ public class ValidateLicensesTest
         validateLicenses.validateLicenses(deps(new Dependency("thing", "1.0", "MIT")), context);
 
         verify(context).newIssue();
-        assertThat(issue.toString(), containsString(LicenseCheckMetrics.LICENSE_CHECK_NOT_ALLOWED_LICENSE_KEY));
+        assertThat(issue.toString(), containsString(LicenseCheckRulesDefinition.RULE_NOT_ALLOWED_LICENSE_KEY));
     }
 
     @Test
@@ -76,7 +75,8 @@ public class ValidateLicensesTest
         SensorContext context = createContext();
 
         validateLicenses.validateLicenses(
-            deps(new Dependency("thing", "1.0", "Apache-2.0"), new Dependency("another", "2.0", "Apache-2.0")),
+            deps(new Dependency("thing", "1.0", "Apache-2.0", LicenseCheckRulesDefinition.LANG_JS),
+                new Dependency("another", "2.0", "Apache-2.0", LicenseCheckRulesDefinition.LANG_JS)),
             context);
 
         verify(context, never()).newIssue();
@@ -127,7 +127,7 @@ public class ValidateLicensesTest
             context);
 
         verify(context).newIssue();
-        assertThat(issue.toString(), containsString(LicenseCheckMetrics.LICENSE_CHECK_NOT_ALLOWED_LICENSE_KEY));
+        assertThat(issue.toString(), containsString(LicenseCheckRulesDefinition.RULE_NOT_ALLOWED_LICENSE_KEY));
     }
 
     @Test
@@ -140,7 +140,7 @@ public class ValidateLicensesTest
         validateLicenses.validateLicenses(deps(new Dependency("thing", "1.0", "(Apache-2.0 AND Apache-1.1)")), context);
 
         verify(context).newIssue();
-        assertThat(issue.toString(), containsString(LicenseCheckMetrics.LICENSE_CHECK_UNLISTED_KEY));
+        assertThat(issue.toString(), containsString(LicenseCheckRulesDefinition.RULE_UNLISTED_KEY));
     }
 
     //  LGPL OR Apache-2.0 AND MIT
@@ -165,7 +165,7 @@ public class ValidateLicensesTest
         validateLicenses.validateLicenses(deps(new Dependency("thing", "1.0", null)), context);
 
         verify(context).newIssue();
-        assertThat(issue.toString(), containsString(LicenseCheckMetrics.LICENSE_CHECK_UNLISTED_KEY));
+        assertThat(issue.toString(), containsString(LicenseCheckRulesDefinition.RULE_UNLISTED_KEY));
     }
 
     @Test
@@ -178,7 +178,7 @@ public class ValidateLicensesTest
         validateLicenses.validateLicenses(deps(new Dependency("thing", "1.0", "Mamamia")), context);
 
         verify(context).newIssue();
-        assertThat(issue.toString(), containsString(LicenseCheckMetrics.LICENSE_CHECK_UNLISTED_KEY));
+        assertThat(issue.toString(), containsString(LicenseCheckRulesDefinition.RULE_UNLISTED_KEY));
     }
 
     @Test

